src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,115 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exeption.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private final Connection connection;
+
+    public BookDaoImpl(Connection connection) {
+        this.connection = connection;
+    }
+
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books "
+                + "(title, price) VALUES (?, ?)";
+        try (PreparedStatement preparedStatement
+                     = connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            preparedStatement.setString(1, book.getTitle());
+            preparedStatement.setBigDecimal(2, book.getPrice());
+            int executeUpdate = preparedStatement.executeUpdate();
+            if (executeUpdate < 1) {
+                throw new DataProcessingException("Can not process");
+            }
+            ResultSet generatedKeys = preparedStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getLong(1));
+
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Cant create new book ", e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String sql = "SELECT * FROM books WHERE id = ?";
+        try (PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                return Optional.of(parseBook(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to find book by id " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        List<Book> books = new ArrayList<>();
+        String sql = "SELECT * FROM books";
+        try (PreparedStatement preparedStatement = connection.prepareStatement(sql)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                books.add(parseBook(resultSet));
+            }
+
+        } catch (SQLException e) {
+            throw new DataProcessingException("Unable to retrieve list of books", e);
```

Comment:
The error message 'Unable to retrieve list of books' could be more descriptive. Consider including SQL state or additional context to help with debugging.